rust_src/src/data.rs

```diff
@@ -597,6 +598,29 @@ pub unsafe extern "C" fn store_symval_forwarding(
     }
 }
 
+unsafe fn update_buffer_defaults(objvar: *const LispObject, newval: LispObject) {
+    // If this variable is a default for something stored
+    // in the buffer itself, such as default-fill-column,
+    // find the buffers that don't have local values for it
+    // and update them.
+    let defaults = LispBufferRef::new(&mut buffer_defaults as *mut Lisp_Buffer);
+    let defaults_as_object = defaults.as_ptr() as *const LispObject;
+    if objvar > defaults_as_object && objvar < defaults_as_object.add(1) {
```

Comment:
In the original, this test is true when objvar points inside the default_buffer object, which is a long struct containing many LispObject variables.

This version, however, tests to see if objvar points between the first and second LispObject of the default_buffer.

Reply:
Thanks. I have not worked with raw pointers much lately.